File: SPOJ/IBIGAME.py
import math

n = int(input())
for _ in range(n):
	a, b = map(int, input().split(' '))
	ans = int(1)
	if a > b:
		ans = a - b
	else:
		ans = b - a
	if ans % 8 == 7 or ans % 8 == 0 or ans % 8 == 1:
		print("Farael")
	else:
		print("Ziya")

File: SPOJ/IBIGAME.java
import java.util.*;
import java.lang.*;

class IBIGAME{
	public static void main(String args[]) {
		Scanner sc = new Scanner(System.in);
		long t = sc.nextLong();
		for (int tc = 0; tc < t; tc++) {
			long a = sc.nextLong();
			long b = sc.nextLong();
			long c = Math.abs(a - b);
			if (c % 8 == 7 || c % 8 == 0 || c % 8 == 1) {
				System.out.println("Farael");
			} else {
				System.out.println("Ziya");
			}
		}
	}
}